overlord/servicestate/servicestate.go

```diff
@@ -73,6 +74,8 @@ func computeExplicitServices(appInfos []*snap.AppInfo, names []string) map[strin
 // serviceControlTs creates "service-control" task for every snap derived from appInfos.
 func serviceControlTs(st *state.State, appInfos []*snap.AppInfo, inst *Instruction) (*state.TaskSet, error) {
 	servicesBySnap := make(map[string][]string, len(appInfos))
+	// note, this may be called with service names of different snaps
+	// XXX: this doesn't support service names from snapctl (i.e. without snap prefix).
```

Comment:
hmm I'm not sure that this is a problem, when I test this locally, I see that both `snapctl start ...` and `snap start ...` show up here as being named `test-snapd-install-hook-runs-svc.svc` without the `snap.*` prefix

Reply:
note that this was tested with master by just printing off inst.Names

Reply:
Hmmm, you're right, I now see I misread getServiceInfos and we actually don't support "snapctl start justservicename", it needs to be snapname or snapname.servicename, in which case this is fine.

Reply:
Removed.